### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by Genateep**

### CODE REVIEW

File: homework3/task01.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                cached['times'] = times
+            else:
+                cached['times'] -= 1
+            if cached['times'] <= 0:
+                return func(*args, **kwargs)
+            return cached[key]
+
+        return decorate
+
+    return cache
```
Comment (icanhardcode) [inline homework3/task01.py]: for ur local tests set ```times=2``` and create a func:
```
@cache(times=2)
def func():
    return input("? ")
```

run this func 5 times (for example).. do u think it's right behavior?

File: homework3/task02.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def mult(func):
+    """Returns time of parallel execution of func"""
+    values = [x for x in range(500)]
+    start = time.time()
+    with Pool(50) as pool:
+        pool.map(func, values)
+    end = time.time()
+    return end - start
```
Comment (icanhardcode) [inline homework3/task02.py]: ur function should rerturn calculations result.. not estimated spent time

File: homework3/task02.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    time.sleep(random.randint(1, 3))
+    data = hashlib.md5(str(value).encode()).digest()
+    return sum(struct.unpack('<' + 'B' * len(data), data))
+
+
+def mult(func):
+    """Returns time of parallel execution of func"""
+    values = [x for x in range(500)]
+    start = time.time()
+    with Pool(50) as pool:
```
Comment (icanhardcode) [inline homework3/task02.py]: not for every processor ```Pool(50)``` would be optimal.. it's better to watch now may cores ur processor has

File: homework3/task04.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+def is_armstrong(number: int) -> bool:
+    """Returns True if given number is Armstrong number.
+    Only positive integers allowed
+    """
+    return number == sum([int(_)**len(str(number)) for _ in str(number)])
```
Comment (icanhardcode) [inline homework3/task04.py]: ```_``` bad name (even) for temporary variable

u should use "_" in other cases

**Review by icanhardcode — CHANGES_REQUESTED** [review]

File: homework3/task02.py (hunk 3 above)
Comment (Genateep) [reply homework3/task02.py]: now I've set the relative number of threads for core experimentally. I have no idea how to define it another way

**Review by icanhardcode — APPROVED** [review]

**Pull request merged**